### ISSUE TITLE
Tensorboard Logger

### ISSUE THREAD

**Issue by anmolsjoshi** [pytorch/ignite#260]
Working on a Tensorboard Handler for Ignite. 

Currently the code does the following
- Creates a graph of the model on initialization of the event handler
- Creates distributions and histograms for the model at the end of each epoch
- Creates distributions and histograms for the gradients to the parameters of the model at the end of each epoch
- Plots metrics related to training and validation (if available) at the end of each epoch

Features I'm working to add are
- View embedding
- Visualize weights of the model

What other features do you think are necessary for this handler? 
@rave78 @vfdev-5

**Comment by anmolsjoshi** [comment]
I am trying to make the event handler general. Looking through examples, these are the variations I can see arising:
- Logging evaluator metrics (separate engine for evaluating) (MNIST)
- Training engine.state.output, this can be a dictionary, a single value or nothing at all (Fast Neural Style, DCGAN)
- A new attribute to engine.state (see RL example)

Any other variations we can think of?

**Comment by anmolsjoshi** [comment]
Just another idea, we could have the TensorBoard event handler as a Base class that people can add functions that will be called at the end of each epoch. For example:
 
- Stylizing an image at the end of each epoch to see improvement
- Feature maps of a single image through convolution layers
- Generating text, audio, images

Thoughts?

**Comment by rave78** [comment]
@anmolsjoshi sorry for my late reply.  I have not started working on TensorBoard.  However, let me know how I can help you

**Comment by anmolsjoshi** [comment]
@rave78 what are your thoughts on the things listed above? any things you disagree with or think need added?

**Comment by anmolsjoshi** [comment]
@vfdev-5 @rave78 Currently facing issues plotting histograms and writing images at the same time. Will post an issue on tensorboardX for workarounds. Let me know if you guys are interested in looking at the code!

**Comment by rave78** [comment]
@anmolsjoshi I think your list of features is complete.
